src/core/manager-dump.h

```diff
@@ -5,8 +5,25 @@
 
 #include "manager.h"
 
-void manager_dump_jobs(Manager *s, FILE *f, const char *prefix);
-void manager_dump_units(Manager *s, FILE *f, const char *prefix);
+void manager_dump_jobs_by_patterns(Manager *s, FILE *f, char **patterns, const char *prefix);
+void manager_dump_units_by_patterns(Manager *s, FILE *f, char **patterns, const char *prefix);
 void manager_dump(Manager *s, FILE *f, const char *prefix);
-int manager_get_dump_string(Manager *m, char **ret);
+int manager_get_dump_string_by_patterns(Manager *m, char **patterns, char **ret);
 void manager_test_summary(Manager *m);
+
+static inline void manager_dump_jobs(Manager *s, FILE *f, const char *prefix) {
+        manager_dump_jobs_by_patterns(s, f, NULL, prefix);
+}
+
+static inline void manager_dump_units(Manager *s, FILE *f, const char *prefix) {
+        manager_dump_units_by_patterns(s, f, NULL, prefix);
+}
+
+static inline void manager_dump_by_patterns(Manager *m, FILE *f, char **patterns, const char *prefix) {
+        manager_dump_units_by_patterns(m, f, patterns, prefix);
+        manager_dump_jobs_by_patterns(m, f, patterns, prefix);
+}
+
+static inline int manager_get_dump_string(Manager *m, char **ret) {
+        return manager_get_dump_string_by_patterns(m, NULL, ret);
+}
```

Comment:
I do not think it is necessary to introduce such variations. Why not simply add `patterns` argument for `manager_dump_jobs()` or friends?

Reply:
Because most of the users don't need to pass a pattern.

Reply:
'users'? Most calls are at manager-dump.c... These are not library function. unlike functions defined in src/basic or src/shared.